### PR TITLE
Change the data type of message type transaction to hex string 

### PR DESCRIPTION
**Pull request by boyeon555**
Change the data type of message type transaction from string to hex string prefixed with '0x'

- Change sample data
- Add unit test
- Add the checker if data is prefixed with '0x' on Message transaction
- Remove the code to convert the data from string to hex string on sending Message transaction
- Update readme
- Update changelog

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -845,7 +845,7 @@ transaction = MessageTransactionBuilder()\
     .step_limit(1000000)\
     .nid(3)\
     .nonce(100)\
-    .data("test")\
+    .data("0x00..0")\
```
Comment (sink772) [inline README.md]: It would be better to use real example like:
```suggestion
    .data("0x74657374")\
```

File: README.md [hunk 2]
```diff
@@ -1001,7 +1001,7 @@ transaction = MessageTransactionBuilder()\
...
```
Comment (sink772) [inline README.md]: Same as above.

File: iconsdk/builder/transaction_builder.py [hunk 3]
```diff
@@ -143,7 +144,11 @@ class MessageTransaction(Transaction):
     def __init__(self, from_: str, to: str, value: int, step_limit: int, nid: int, nonce: int, version: int,
                  timestamp: int, data: str):
         Transaction.__init__(self, from_, to, value, step_limit, nid, nonce, version, timestamp)
-        self.__data = data
+
+        if is_0x_prefixed(data):
+            self.__data = data
+        else:
+            raise DataTypeException("Message's data should be hex string prefixed with '0x'.")
```
Comment (sink772) [inline iconsdk/builder/transaction_builder.py]:
```suggestion
            raise DataTypeException("Message data should be hex string prefixed with '0x'.")
```

File: README.md (hunk 1 above)
Comment (boyeon555) [reply README.md]: Applied.

File: tests/api_send/test_send_message.py [hunk 4]
```diff
@@ -42,12 +44,37 @@ def test_send_message(self):
         self.assertTrue(is_T_HASH(result))
 
         # When having an optional property, nonce
+        sleep(2)
```
Comment (leeheonseung) [inline tests/api_send/test_send_message.py]: It's option

I think that
before : sleep(2)

somewhere : 
sleep_time = 2
after : sleep(sleep_time)

File: tests/api_send/test_send_super.py [hunk 5]
```diff
@@ -74,7 +74,7 @@ def setUpClass(cls):
                 "init_supply": 10000
             },
             # It is used to send message only.
-            "data": "test"
+            "data": "0x" + "test".encode().hex()
```
Comment (leeheonseung) [inline tests/api_send/test_send_super.py]: I think that

before : encode()
after : encode(utf-8)

Comment (boyeon555) [reply tests/api_send/test_send_super.py]: But does not it work as same? Default is utf-8.

File: iconsdk/builder/transaction_builder.py [hunk 6]
```diff
@@ -143,7 +144,11 @@ class MessageTransaction(Transaction):
     def __init__(self, from_: str, to: str, value: int, step_limit: int, nid: int, nonce: int, version: int,
                  timestamp: int, data: str):
         Transaction.__init__(self, from_, to, value, step_limit, nid, nonce, version, timestamp)
-        self.__data = data
+
+        if is_0x_prefixed(data):
```
[the remark this answers is not in the document]
Comment (boyeon555) [reply iconsdk/builder/transaction_builder.py]: Applied. Thank you.

**Pull request merged**